cromshell

```diff
@@ -1471,6 +1488,99 @@ function fetch-all()
   return 0
 }
 
+# Get the cost for a workflow ID:
+function cost()
+{
+  which bq &>/dev/null   
+  local r=$?
+  [[ ${r} -ne 0 ]] && error "bq does not exist.  Must install the big query command-line client." && exit 8
+
+  [ ! -e ${BQ_COST_TABLE_FILE} ] && error "Big Query cost table file does not exist.  Must populate ${BQ_COST_TABLE_FILE} with big query cost table information." && exit 9
+
+  local cost_table=$(head -n1 ${BQ_COST_TABLE_FILE})
+  local id=$1
+
+  if [[ "$(uname)" == "Darwin" ]] ; then
+    local DATE_CMD=gdate
+  else
+    local DATE_CMD=date
+  fi  
+
+  local submissionDate=$( grep "${id}" ${CROMWELL_SUBMISSIONS_FILE} | head -n1 | awk '{print $1}' )
+  local y=$( echo ${submissionDate} | sed 's#^\([0-9][0-9][0-9][0-9]\).*#\1#' )
+  local m=$( echo ${submissionDate} | sed 's#^[0-9][0-9][0-9][0-9]\([0-9][0-9]\).*#\1#' )
+  local d=$( echo ${submissionDate} | sed 's#^[0-9][0-9][0-9][0-9][0-9][0-9]\([0-9][0-9]\).*#\1#' )
+
+  local START_DATE="${y}-${m}-${d}"
```

Comment:
I think the max PAPI runtime is 7 days so maybe that instead?

Reply:
Doing the queries to find out how much a workflow cost actually costs money, so I wanted to keep the queries as small as possible.  I feel like I'm pushing it even at 4 days.  If and when I implement the metadata checks for the actual end time, I may use that instead.

For now I don't want to change it (in an effort to preemptively save costs).